### PR TITLE
Clang tidy CI test: add performance-unnecessary-* checks to clang-tidy CI test

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR adds the following checks to clang-tidy CI test:

- [performance-unnecessary-copy-initialization](https://releases.llvm.org/14.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/performance-unnecessary-copy-initialization.html)
- [performance-unnecessary-value-param](https://releases.llvm.org/14.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/performance-unnecessary-value-param.html)

I will continue working on this PR after https://github.com/ECP-WarpX/WarpX/pull/4133 is merged, since it is likely to fix several of the issues found with `performance-unnecessary-value-param` check

### CODE REVIEW

**Comment by ax3l** [comment]
@lucafedeli88 this needs a rebase, but I would merge it after #3474 is in

**Comment by ax3l** [comment]
@lucafedeli88 sorry, can you rebase pls? :)
I merged #4146

**Comment by lucafedeli88** [comment]
> @lucafedeli88 sorry, can you rebase pls? :) I merged #4146

@ax3l , I should have fixed the conflicts!

**Comment by ax3l** [comment]
Ohoh, CI says no @lucafedeli88

File: Source/Diagnostics/BTDiagnostics.H [hunk 1]
```diff
@@ -375,26 +375,26 @@ private:
...
-                                             std::string snapshot_Header);
+    void InterleaveBufferAndSnapshotHeader (const std::string& buffer_Header,
+                                            const std::string& snapshot_Header);
     /** Interleave meta-data of the buffer multifabs to be consistent
      *  with the merged plotfile lab-frame data.
      */
-    void InterleaveFabArrayHeader (std::string Buffer_FabHeader_path,
-                                   std::string snapshot_FabHeader_path,
-                                   std::string newsnapshot_FabFilename);
+    void InterleaveFabArrayHeader(const std::string& Buffer_FabHeader_path,
```
Comment (dpgrote) [inline Source/Diagnostics/BTDiagnostics.H]: This is missing the space before the parenthesis
```suggestion
    void InterleaveFabArrayHeader (const std::string& Buffer_FabHeader_path,
```

Comment (dpgrote) [reply Source/Diagnostics/BTDiagnostics.H]: Though I see many other places without the space, so probably doesn't need to be fixed. It's a nice detail but used inconsistently.

**Review by dpgrote — APPROVED** [review]
The changes look good to me! Thanks!

**Pull request merged**